src/ui/server/helpers/summary-lists/generate-field-group-item/index.ts

```diff
@@ -14,11 +15,21 @@ import { SummaryListItemData, SummaryListItemDataInput } from '../../../../types
  */
 export const getSummaryListItemDataValue = (fieldId: string, data?: object, customValue?: string): string => {
   if (customValue) {
-    return customValue;
+    return String(customValue);
   }
 
   if (data && objectHasProperty(data, fieldId)) {
-    return data[fieldId];
+    return String(data[fieldId]);
+  }
+
+  /**
+   * if data is the number 0, then transform to string
+   * 0 fails the objectHasProperty(data, fieldId) as read as false
+   */
+  if (data && transformNumberToString(data[fieldId])) {
+    const value = transformNumberToString(data[fieldId]);
+
+    return String(value);
```

Comment:
Could this be simpler like this?

```js

if (data) {
  if (objectHasProperty(data, fieldId)) {
    return String(data[fieldId]);
  }

  return transformNumberToString(data[fieldId]);
}
```

If `transformNumberToString` returns a string shouldn't have to wrap with `String`)

Reply:
eslint does not like it as the function could return undefined/null

Reply:
also I need to have the if transform to number as it will never reach the return DEFAULT.EMPTY; and will not display the -